MantidQt/CustomInterfaces/src/EnggDiffraction/EnggDiffFittingPresenter.cpp

```diff
@@ -0,0 +1,1126 @@
+
+  try {
+    std::string cwd(bankDir);
+    Poco::DirectoryIterator it(cwd);
+    Poco::DirectoryIterator end;
+    while (it != end) {
+      if (it->isFile()) {
+        std::string itFilePath = it->path();
+        Poco::Path itBankfPath(itFilePath);
+
+        std::string itbankFileName = itBankfPath.getBaseName();
+        // check if it not any other file.. e.g: texture
+        if (itbankFileName.find(focusedFile) != std::string::npos) {
+          fittingRunNoDirVec.push_back(itFilePath);
+          if (multi_run)
+            return;
+        }
+      }
+      ++it;
+    }
+  } catch (std::runtime_error &re) {
+    m_view->userWarning("Invalid file",
+                        "File not found in the following directory; " +
+                            bankDir + ". " +
+                            static_cast<std::string>(re.what()));
+  }
+}
+
+void EnggDiffFittingPresenter::enableMultiRun(
+    std::string firstRun, std::string lastRun,
+    std::vector<std::string> &fittingRunNoDirVec) {
+
+  bool firstDig = isDigit(firstRun);
+  bool lastDig = isDigit(lastRun);
+
+  std::vector<std::string> RunNumberVec;
+  if (firstDig && lastDig) {
+    int firstNum = std::stoi(firstRun);
+    int lastNum = std::stoi(lastRun);
+
+    if ((lastNum - firstNum) > 200) {
+      m_view->userWarning(
+          "Please try again",
+          "The specified run number range is "
+          "far to big, please try a smaller range of consecutive run numbers.");
+    }
+
+    else if (firstNum <= lastNum) {
+
+      for (int i = firstNum; i <= lastNum; i++) {
+        RunNumberVec.push_back(std::to_string(i));
+      }
+
+      auto focusDir = m_view->focusingDir();
+      if (focusDir.empty()) {
+        m_view->userWarning(
+            "Invalid Input",
+            "Please check that a valid directory is "
+            "set for Output Folder under Focusing Settings on the "
+            "settings tab. "
+            "Please try again");
+      } else {
+        // if given a single run number instead
+        for (size_t i = 0; i < RunNumberVec.size(); i++) {
+          updateFittingDirVec(focusDir, RunNumberVec[i], true,
+                              fittingRunNoDirVec);
+        }
+        int diff = (lastNum - firstNum) + 1;
+        auto global_vec_size = fittingRunNoDirVec.size();
+        if (size_t(diff) == global_vec_size) {
+
+          setRunNoItems(RunNumberVec, true);
+
+          m_view->setBankEmit();
+        }
+      }
+    } else {
+      m_view->userWarning("Invalid Run Number",
+                          "One or more run file not found "
+                          "from the specified range of runs."
+                          "Please try again");
+    }
+  } else {
+    m_view->userWarning("Invalid Run Number",
+                        "The specfied range of run number "
+                        "entered is invalid. Please try again");
+  }
+}
+
+void EnggDiffFittingPresenter::processStart() {}
```

Comment:
Yes. For example, at present you only get a preview plot after you run `Fit`. But a preview of the raw data before fitting is useful. The tab should show something on the preview plot if a run number is already selected (either in the current session or saved from previous sessions), without waiting for the user to run Fit. Functionality like that should be controlled/triggered from here.